src/steps/data_validation/utils/dataset_validator.py

```diff
@@ -0,0 +1,145 @@
+    dataset_context: DatasetContext, image_path_list: list[str]
+) -> None:
+    """
+    Checks for corruption in the extracted images.
+
+    Parameters:
+        dataset_context (DatasetContext): The dataset context to validate.
+        image_path_list (list[str]): The list of image paths extracted from the dataset.
+
+    Raises:
+        ValueError: If any of the images are found to be corrupted.
+    """
+    for image_path in image_path_list:
+        try:
+            with Image.open(image_path) as img:
+                img.verify()  # Verify that this is a valid image
+        except Exception as e:
+            raise ValueError(
+                f"Image {image_path} is corrupted in {dataset_context.dataset_name} dataset - {e}"
+            )
+
+
+class DatasetValidator:
+    """
+    Validates various aspects of datasets within a dataset collection.
+
+    This class performs common validation tasks for datasets, including checking for image extraction
+    completeness, image format, image corruption, and annotation integrity. It serves as a base class
+    for more specific dataset validators.
+
+    Attributes:
+        dataset_collection (DatasetCollection): The collection of datasets to validate.
+    """
+
+    VALID_IMAGE_EXTENSIONS = (".jpg", ".jpeg", ".png")
+
+    def __init__(self, dataset_collection: DatasetCollection):
+        """
+        Initializes the DatasetValidator with a dataset collection to validate.
+
+        Parameters:
+            dataset_collection (DatasetCollection): The dataset collection to validate.
+        """
+        self.dataset_collection = dataset_collection
+
+    def validate_common(self) -> None:
+        """
+        Performs common validation tasks across all datasets in the collection.
+        """
+        for dataset_context in self.dataset_collection:
+            image_path_list = get_image_path_list(image_dir=dataset_context.image_dir)
+            validate_image_extraction(
+                dataset_context=dataset_context, image_path_list=image_path_list
+            )
+            self.validate_image_format(
+                dataset_context=dataset_context, image_path_list=image_path_list
+            )
+            validate_image_corruption(
+                dataset_context=dataset_context, image_path_list=image_path_list
+            )
+            self.validate_image_annotation_integrity(dataset_context)
+
+    def validate_image_format(self, dataset_context, image_path_list) -> None:
+        """
+        Validates that all images in the dataset are in an expected format.
+
+        Args:
+            dataset_context (DatasetContext): The dataset context to validate.
+            image_path_list (list[str]): The list of image paths extracted from the dataset.
+
+        Raises:
+            ValueError: If any image is not in one of the valid formats.
+        """
+        for image_path in image_path_list:
+            if not image_path.endswith(self.VALID_IMAGE_EXTENSIONS):
```

Comment:
you can apply a `lower()` function since the `VALID_IMAGE_EXTENSIONS` also exist in uppercase. Something like:

```python
(...)
if not image_path.lower().endswith(self.VALID_IMAGE_EXTENSIONS):
(...)
```